drivers/frequency/ad9517/ad9517.c

```diff
@@ -332,9 +333,30 @@ int64_t ad9517_vco_frequency(struct ad9517_dev *dev,
 	uint8_t prescaler = 0;
 	uint32_t reg_value = 0;
 
-	if((frequency < AD9517_MIN_VCO_FREQ) ||
-	    (frequency > AD9517_MAX_VCO_FREQ))
+	switch(dev->ad9517_type) {
+	case AD9517_1:
+		if((frequency < AD9517_1_MIN_VCO_FREQ) ||
+		    (frequency > AD9517_1_MAX_VCO_FREQ))
+			return -1;
```

Comment:
use `FAILURE`, it will be easier to search and replace it when we will have more detailed return error codes

Reply:
The driver does not include `error.h`.
This change is related only to adding part types.
Including SUCCESS and FAILURE should be part of another commit.
Moreover, the current state of the driver includes functions with return values that are different from SUCCESS/FAILURE return codes.

Reply:
ok, sure ! will approve like this